pod/SynthVoice/PodSynthVoice.cpp

```diff
@@ -0,0 +1,163 @@
+// Simple Synth voice example. Turning the encoder cycles through three control modes. Each mode has different knob functions.
+// In mode one (blue), knob one controls filter cutoff and knob two controls oscillator frequency.
+// In mode two (green), knob one 1 controls envelope attack time and knob two controls envelope decay time.
+// In mode three (red), knob one controls vibrato rate and knob two controls vibrato depth. Very fast vibratos are heard as simple FM synthesis!
+// Click the encoder to cycle through waveforms. Press either button to trigger the envelope.
+// Leds light to indicate what mode is selected.
+
+#include "daisysp.h"
+#include "daisy_pod.h"
+
+using namespace daisysp;
+using namespace daisy;
+
+static DaisyPod pod;
+static Oscillator osc, lfo;
+static MoogLadder flt;
+static AdEnv ad;
+static Parameter pitch, cutoff, lfof;
+
+int wave, mode;
+float v, p;
+float oldk1, oldk2;
+
+static void AudioCallback(float *in, float *out, size_t size)
+{
+    float sig, c, ad_out, attack, release, lf, la;
+
+    pod.UpdateAnalogControls();
+    pod.DebounceControls();
+    
+    wave += pod.encoder.RisingEdge();
+    wave %= osc.WAVE_LAST;
+    osc.SetWaveform(wave);
+    
+    mode += pod.encoder.Increment();
+    mode = (mode % 3 + 3) % 3;
+
+    float k1 = pod.knob1.Process();
+    float k2 = pod.knob2.Process();
+
+    //knobs
+    if (abs(oldk1 - k1) > 0.0005 || abs(oldk2 - k2) > 0.0005){
+        switch (mode)
+	{
+	    case 0:
+	        p  = pitch.Process();
+		c = cutoff.Process();
+		flt.SetFreq(c);
+		break;
+	    case 1:
+  	        attack   = pod.GetKnobValue(pod.KNOB_1);
+		release  = pod.GetKnobValue(pod.KNOB_2);
+		ad.SetTime(ADENV_SEG_ATTACK, attack);
+		ad.SetTime(ADENV_SEG_DECAY, release);
+		break;
+	    case 2:
+  	        lf = lfof.Process();
+		la = pod.GetKnobValue(pod.KNOB_2);
+		lfo.SetFreq(lf * 500);
+		lfo.SetAmp(la * 100);
+	    default:
+	        break;
+	}
+    }
+
+    //leds
+    switch (mode)
+    {
+        case 0:
+	    pod.led1.Set(0,0,1);
+	    pod.led2.Set(0,0,1);
+	    break;
+        case 1:
+	    pod.led1.Set(0,1,0);
+	    pod.led2.Set(0,1,0);
+	    break;
+        case 2:
+	    pod.led1.Set(1,0,0);
+	    pod.led2.Set(1,0,0);
+        default: 
+	    break;
+    }
```

Comment:
might be a little cleaner looking like:
```
switch(mode)
{
    case N:
        if (change)
            setparams;
        pod.ledN.Set(mode_color);
        break;
}
````

Even though its more times doing the check, it gets rid of a whole switch statement, which in my opinion made it feel like a much longer example.

Reply:
Better yet, you could set up a `CheckChange(current, previous);` function and have a check-per parameter. 

I didn't realize it while I was playing but once you turned one knob both parameters take over.